src/samplers/cpu/mod.rs

```diff
@@ -149,6 +120,74 @@ impl Sampler for Cpu {
 }
 
 impl Cpu {
+    #[cfg(feature = "bpf")]
+    fn bpf_enabled(&self) -> bool {
+        if self.sampler_config().bpf() {
+            for statistic in self.sampler_config().statistics() {
+                if statistic.bpf_table().is_some() {
+                    return true;
+                }
+            }
+        }
+        false
+    }
+
+    fn initialize_bpf(&mut self) -> Result<(), failure::Error> {
+        #[cfg(feature = "bpf")]
+        {
+            if self.enabled() && self.bpf_enabled() {
+                debug!("initializing bpf");
+                let code = include_str!("bpf.c");
+                let mut bpf = bcc::core::BPF::new(code)?;
+
+                for statistic in self.sampler_config().statistics() {
+                    if let Some(config) = statistic.bpf_config() {
+                        PerfEventProbe::new()
+                            .name(table.0)
+                            .event(table.1)
+                            .sample_period(SAMPLE_PERIOD)
+                            .attach(&mut bpf);
+                    }
+                }
+
+                self.bpf = Some(Arc::new(Mutex::new(BPF { inner: bpf })))
+            }
+        }
+        Ok(())
+    }
+
+    #[cfg(feature = "bpf")]
+    fn sample_bpf(&self) -> Result<(), std::io::Error> {
+        if self.bpf_last.lock().unwrap().elapsed() >= self.general_config().window() {
+            if let Some(ref bpf) = self.bpf {
+                let bpf = bpf.lock().unwrap();
+                let time - time::precise_time_ns();
+
+                for statistic in self.sampler_config().statistics() {
+                    if let Some(config) = statistics.bpf_config() {
```

Comment:
Maybe something like this? `if let Some((name, event)) = `

Reply:
You either need exactly one of sample_period or sample_frequency. Maybe set it to 1?